### PR TITLE
Added unit testing to github-mirror

### PR DESCRIPTION
**Pull request by msk999**
85% code coverage from added unit tests.  Added various gems to allow for mocks and stubs and HTTP playback.  Please let us know what you think.

### CODE REVIEW

**Comment by gousiosg** [comment]
Thanks! This is a big one! Currently inspecting...

File: test/ghtorrent_test.rb [hunk 1]
```diff
@@ -0,0 +1,75 @@
+require 'test_helper'
+
+class GhtorrentTest
+  describe 'test configuration and helper methods' do
+    before do
+     @ght = ght
+     @db = db
+    end
+
+   it 'should be able to access configurations' do
```
Comment (gousiosg) [inline test/ghtorrent_test.rb]: This is perhaps excessive

**Comment by msk999** [comment]
Did further clean up of the unit test code.  A change was made to the test factories to ensure the tests worked in PostgresQL environment.   Incorporated your changes and restored Sqlite3 to be the default database for testing purposes.  

Ran the unit tests agains Sqlite3, MySql and PostgreSQL and all the tests are passing.   

Please let us know if you have any further comments.  The changes for eTags will be coming shortly.

**Comment by gousiosg** [comment]
The results above where on my laptop (Mac). On Ubuntu 16.04.4 LTS, I get even worse results:

```
265 tests, 0 assertions, 1 failures, 259 errors, 0 skips
```

Trace: https://pastebin.com/wCMVt0gM

What environment are you using to run the tests?

**Comment by msk999** [comment]
We are using Ruby 2.2.4.

**Comment by gousiosg** [comment]
Ubuntu 16.04 LTS uses 2.3 by default

```
$ ruby -v
ruby 2.3.1p112 (2016-04-26) [x86_64-linux-gnu]
```

2.2 has been EOLed: https://www.ruby-lang.org/en/news/2018/06/20/support-of-ruby-2-2-has-ended/

**Comment by gousiosg** [comment]
Thanks and sorry it took ages to resolve

**Pull request merged**